gsa/src/web/pages/scanconfigs/__tests__/listpage.js

```diff
@@ -0,0 +1,243 @@
+/* Copyright (C) 2019 Greenbone Networks GmbH
+ *
+ * SPDX-License-Identifier: GPL-2.0-or-later
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 2
+ * of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ */
+import React from 'react';
+
+import Capabilities from 'gmp/capabilities/capabilities';
+import CollectionCounts from 'gmp/collection/collectioncounts';
+
+import Filter from 'gmp/models/filter';
+import ScanConfig, {
+  OPENVAS_SCAN_CONFIG_TYPE,
+  SCANCONFIG_TREND_STATIC,
+  SCANCONFIG_TREND_DYNAMIC,
+} from 'gmp/models/scanconfig';
+
+import {setUsername} from 'web/store/usersettings/actions';
+
+import {entitiesActions} from 'web/store/entities/scanconfigs';
+import {loadingActions} from 'web/store/usersettings/defaults/actions';
+import {defaultFilterLoadingActions} from 'web/store/usersettings/defaultfilters/actions';
+
+import {rendererWith, waitForElement, fireEvent} from 'web/utils/testing';
+
+import ScanConfigsPage, {ToolBarIcons} from '../listpage';
+
+const config = ScanConfig.fromElement({
+  _id: '12345',
+  name: 'foo',
+  comment: 'bar',
+  creation_time: '2019-07-16T06:31:29Z',
+  modification_time: '2019-07-16T06:44:55Z',
+  owner: {name: 'admin'},
+  writable: '1',
+  in_use: '0',
+  usage_type: 'scan',
+  permissions: {permission: [{name: 'everything'}]},
+  scanner: {name: 'scanner', type: '42'},
+  type: OPENVAS_SCAN_CONFIG_TYPE,
+  tasks: {
+    task: [{id: '1234', name: 'task1'}, {id: '5678', name: 'task2'}],
+  },
+  family_count: {
+    __text: 2,
+    growing: SCANCONFIG_TREND_STATIC,
+  },
+  nvt_count: {
+    __text: 4,
+    growing: SCANCONFIG_TREND_DYNAMIC,
+  },
+});
+
+const caps = new Capabilities(['everything']);
+const wrongCaps = new Capabilities(['get_config']);
+
+const reloadInterval = 1;
+const manualUrl = 'test/';
+
+const currentSettings = jest.fn().mockReturnValue(
```

Comment:
You could simplify these calls by using [mockResolvedValue](https://jestjs.io/docs/en/mock-function-api#mockfnmockresolvedvaluevalue)